src/server/scripts/Northrend/ChamberOfAspects/RubySanctum/boss_halion.cpp

```diff
@@ -1473,6 +1491,47 @@ class spell_halion_combustion_consumption : public SpellScriptLoader
         uint32 _spellID;
 };
 
+class spell_halion_combustion_consumption_periodic : public SpellScriptLoader
+{
+    public:
+        spell_halion_combustion_consumption_periodic() : SpellScriptLoader("spell_halion_combustion_consumption_periodic") { }
+
+        class spell_halion_combustion_consumption_periodic_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_halion_combustion_consumption_periodic_AuraScript);
+
+            bool Validate(SpellInfo const* spellInfo) override
+            {
+                if (!sSpellMgr->GetSpellInfo(spellInfo->Effects[EFFECT_0].TriggerSpell))
+                    return false;
+                return true;
+            }
+
+            void HandleTick(AuraEffect const* aurEff)
+            {
+                PreventDefaultAction();
+                Unit* caster = GetCaster();
+                if (!caster)
+                    return;
+
+                uint32 triggerSpell = GetSpellInfo()->Effects[aurEff->GetEffIndex()].TriggerSpell;
+                int32 radius = caster->GetObjectScale() * M_PI * 10000 / 3;
```

Comment:
nothing, trial and error until you find something that fits the visual range of the spell